docs/resources/keycloak_group.md

```diff
@@ -0,0 +1,55 @@
+# keycloak_group
+
+Allows for creating and managing Groups within Keycloak.
+
+Groups provide a logical wrapping for users within Keycloak. Users within a
+group can share attributes and roles, and group membership can be mapped
+to a claim.
+
+Groups can also be federated from external data sources, such as LDAP or Active Directory.
+This resource **should not** be used to manage groups that were created this way.
+
+### Example Usage
+
+```hcl
+resource "keycloak_realm" "realm" {
+    realm   = "my-realm"
+    enabled = true
+}
+
+resource "keycloak_group" "parent_group" {
+    realm_id = "${keycloak_realm.realm.id}"
+    name     = "parent-group"
+}
+
+resource "keycloak_group" "child_group" {
+    realm_id  = "${keycloak_realm.realm.id}"
+    parent_id = "${keycloak_group.parent_group.id}"
+    name      = "child-group"
+}
+```
+
+### Argument Reference
+
+The following arguments are supported:
+
+- `realm_id` - (Required) The realm this groups exists in.
```

Comment:
Thoughts on auto-generating the arguments section by reading the TF schema and extracting descriptions?

Reply:
I considered this, but decided it would probably be a lot of work to do, especially for resources or attributes that have a lot of configuration options.

I would be totally open to trying something out if you have any ideas though!